proof-systems/src/groth16/benches/bn382_gro16_test_circuits.rs

```diff
@@ -0,0 +1,248 @@
+use algebra::{
+    fields::bn_382::Fr,
+    curves::bn_382::Bn382,
+    UniformRand, PrimeField,
+};
+use r1cs_core::{SynthesisError, ConstraintSynthesizer, ConstraintSystem};
+use proof_systems::groth16::{generate_random_parameters, create_random_proof};
+
+use criterion::{BenchmarkId, BatchSize};
+use criterion::Criterion;
+use r1cs_std::Assignment;
+use r1cs_std::fields::fp::FpGadget;
+use r1cs_std::eq::EqGadget;
+use r1cs_std::fields::FieldGadget;
+use r1cs_std::alloc::AllocGadget;
+
+use rand::{
+    rngs::OsRng, thread_rng
+};
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+#[macro_use]
+extern crate criterion;
+
+#[macro_use]
+extern crate bench_utils;
+
+pub struct TestCircuit1<F: PrimeField> {
+    num_constraints: usize,
+    a: Option<F>,
+    b: Option<F>,
+}
+
+pub struct TestCircuit2<F: PrimeField> {
+    num_constraints: usize,
+    a: Option<F>,
+    b: Option<F>,
+}
+
+impl<F: PrimeField> ConstraintSynthesizer<F> for TestCircuit1<F> {
```

Comment:
We should comment that TestCircuit1 has R1CS density d=1 and keeps the synthesizer costs low (no field inversion needed).